### PR TITLE
master: Add python_requires to help pip

### PR DESCRIPTION
**Pull request by hugovk**
#### References to other Issues or PRs

Helps towards #17943 "Removing Python 2 code".

Like https://github.com/sympy/sympy/pull/17966 but for `master`.

#### Brief description of what is fixed or changed

Add `python_requires` to help pip install the right version of this package for the user's running Python.

Also update the Trove classifiers.

Also update the version check in `setup.py`, Python 3.4 was already dropped in https://github.com/sympy/sympy/pull/16492. This is not strictly necessary with `python_requires`, unless the user has a really old version of pip and could probably be removed entirely.

#### Release Notes

[I'm not sure if this requires user-facing release notes, so didn't add any, but please let me know if otherwise.]

NO ENTRY

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
This looks good but this won't be mergeable until we drop Python 2.7 from Travis. I think we've agreed to hold off on that for a little while just until we get feedback from the SymPy 1.5 release.

File: sympy/__init__.py [hunk 1]
```diff
@@ -35,10 +35,8 @@ def enable_warnings():
 
 
 import sys
-if ((sys.version_info[0] == 2 and sys.version_info[1] < 7) or
-    (sys.version_info[0] == 3 and sys.version_info[1] < 4)):
-    raise ImportError("Python version 2.7 or 3.4 or above "
-                      "is required for SymPy.")
+if sys.version_info < (3, 5):
+    raise ImportError("Python 3.5 or above is required for SymPy.")
```
Comment (kenluck2001) [inline sympy/__init__.py]: Are you making a backward-incompatible change?

Comment (hugovk) [reply sympy/__init__.py]: Yes, the idea is to drop support for Python 2.7: https://github.com/sympy/sympy/issues/17943.

**Comment by hugovk** [comment]
Rebased to resolve conflicts.

File: sympy/__init__.py (hunk 1 above)
Comment (asmeurer) [reply sympy/__init__.py]: 1.5.1 will be the last version to support Python 2. See https://github.com/sympy/sympy/wiki/Python-version-support-policy

**Comment by oscarbenjamin** [comment]
I guess this needs a change to the `TEST_PY2_IMPORT` case in `test_travis.sh`. Maybe we can test importing sympy before trying to install it.

**Comment by oscarbenjamin** [comment]
I have a fix for this in #18244 (changes to `.travis.yml` and `bin/test_py2_import`). That fix is needed or we also can't have any Py3 syntax in the main codebase...

**Comment by oscarbenjamin** [comment]
It might make sense to extract those changes and push them here

**Comment by hugovk** [comment]
Thanks, updated, and `TEST_PY2_IMPORT` now passes!

Also removed the redundant `sudo:` lines from `.travis.yml`.

* https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

**Comment by czgdp1807** [comment]
@hugovk Please resolve this conflicts. Probably after that it should be good to go as can be concluded from the the above comments.

**Comment by hugovk** [comment]
@czgdp1807 Conflict resolved!

**Comment by asmeurer** [comment]
What is the purpose of the `sudo` changes in .travis.yml?

Other than that, the rest looks good to me.

**Comment by hugovk** [comment]
@asmeurer `sudo` no longer has any effect:

* [blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration](https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration)

**Comment by asmeurer** [comment]
I guess it works without it.

**Pull request merged**